src/engine/blocks.js

```diff
@@ -663,6 +683,10 @@ class Blocks {
         }
         }
 
+        // TODO maybe track actual changes,
```

Comment:
Is this super hard to figure out in this pass?

Reply:
Ah I had actually put this in here as a note to myself to see whether we needed to track actual changes or if this was good enough. I think in general it is usually rare that we would receive a `change` event from scratch-blocks where a change did not occur. This is unlike the other cases (create, move, delete) where we get the respective events from scratch-blocks but effectively no change occurs in the VM.

I don't *think* it's possible to get a change event for a block without some user interaction with the block. @picklesrus, I would be interested to know, can you think of scenarios where this would happen?

I think I'll probably remove this TODO comment.